conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -1321,6 +1324,197 @@ TEST_F(
+
+    ze_device_properties_t deviceProperties = {
+        ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES, nullptr};
+    zeDeviceGetProperties(device, &deviceProperties);
+
+    LOG_INFO << "test device name " << deviceProperties.name << " uuid "
+             << lzt::to_string(deviceProperties.uuid);
+    if (deviceProperties.flags & ZE_DEVICE_PROPERTY_FLAG_SUBDEVICE) {
+      LOG_INFO << "test subdevice id " << deviceProperties.subdeviceId;
+    } else {
+      LOG_INFO << "test device is a root device";
+    }
+
+    ze_command_queue_handle_t commandQueue = lzt::create_command_queue(device);
+    zet_command_list_handle_t commandList = lzt::create_command_list(device);
+
+    auto metricGroupInfo = lzt::get_metric_group_info(
+        device, ZET_METRIC_GROUP_SAMPLING_TYPE_FLAG_TIME_BASED, true);
+    metricGroupInfo = lzt::optimize_metric_group_info_list(metricGroupInfo);
+
+    void *a_buffer, *b_buffer, *c_buffer;
+    ze_group_count_t tg;
+    ze_kernel_handle_t function = get_matrix_multiplication_kernel(
+        device, &tg, &a_buffer, &b_buffer, &c_buffer, 8192);
+    zeCommandListAppendLaunchKernel(commandList, function, &tg, nullptr, 0,
+                                    nullptr);
+    lzt::close_command_list(commandList);
+
+    for (auto groupInfo : metricGroupInfo) {
+      LOG_INFO << "test metricGroup name " << groupInfo.metricGroupName;
+      lzt::activate_metric_groups(device, 1, &groupInfo.metricGroupHandle);
+
+      ze_event_handle_t eventHandle;
+      lzt::zeEventPool eventPool;
+      eventPool.create_event(eventHandle, ZE_EVENT_SCOPE_FLAG_HOST,
+                              ZE_EVENT_SCOPE_FLAG_HOST);      
+
+      auto startTime = std::chrono::system_clock::now();
+      zet_metric_streamer_handle_t metricStreamerHandle =
+          lzt::metric_streamer_open_for_device(
+              device, groupInfo.metricGroupHandle, eventHandle,
+              notifyEveryNReports, samplingPeriod);
+      ASSERT_NE(nullptr, metricStreamerHandle);
+
+      lzt::execute_command_lists(commandQueue, 1, &commandList, nullptr);
+      lzt::synchronize(commandQueue, std::numeric_limits<uint64_t>::max());
+
+      double minimumTimeBeforeEventIsExpected = notifyEveryNReports * (static_cast<double>(samplingPeriod) / 1000000000);
+      // Initializing a 5% error buffer to prevent corner cases 
+      double errorBuffer = 0.05 * minimumTimeBeforeEventIsExpected;
+      LOG_INFO << "minimumTimeBeforeEventIsExpected " << minimumTimeBeforeEventIsExpected;
+      LOG_INFO << "errorBuffer " << errorBuffer;
+      
+      ze_result_t eventResult;
+      while (true) {
+        auto currentTime = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsedSeconds = currentTime - startTime;
+        if (elapsedSeconds.count() >= (minimumTimeBeforeEventIsExpected - errorBuffer)) {
+          break;
+        }
+        eventResult = zeEventQueryStatus(eventHandle);
+        EXPECT_EQ(eventResult, ZE_RESULT_NOT_READY);
```

Comment:
No we are subtracting the buffer time here as well to avoid such cases.

`if (elapsedSeconds.count() >= (minimumTimeBeforeEventIsExpected - errorBuffer)) {`